listeners.go

```diff
@@ -87,12 +87,24 @@ func ListenPacket(network, addr string) (net.PacketConn, error) {
 
 // ListenQUIC returns a quic.EarlyListener suitable for use in a Caddy module.
 // Note that the context passed to Accept is currently ignored, so using
-// a context other than context.Background is meaningless.
-func ListenQUIC(addr string, tlsConf *tls.Config) (quic.EarlyListener, error) {
+// a context other than context.Background is meaningless. If activeRequests
+// is not nil, it should represent the current load on the server.
+func ListenQUIC(addr string, tlsConf *tls.Config, activeRequests *int64) (quic.EarlyListener, error) {
 	lnKey := "quic/" + addr
 
 	sharedEl, _, err := listenerPool.LoadOrNew(lnKey, func() (Destructor, error) {
-		el, err := quic.ListenAddrEarly(addr, http3.ConfigureTLSConfig(tlsConf), &quic.Config{})
+		el, err := quic.ListenAddrEarly(addr, http3.ConfigureTLSConfig(tlsConf), &quic.Config{
+			AcceptToken: func(clientAddr net.Addr, token *quic.Token) bool {
```

Comment:
General explanation: When sending the first Initial packet to an unknown server, the client doesn't have any token, and thus won't send any. If the server decides to perform a Retry (which costs 1 RTT), it sends a Retry packet containing a token, which the client will then use on all subsequent Initial packets.
For reconnects to a server there's another case one needs to consider: During the connection, the server provides tokens to the client to use on future connections.
When receiving (and validating a token), a server probably wants to accept tokens provided in Retry packets for only a short time frame (10s or so), while tokens provided on a fully established connection would be valid for longer. The [quic.Token](https://pkg.go.dev/github.com/lucas-clemente/quic-go#Token) allows you to distinguish between these two cases.

You probably therefore want to:
1. First check if the server is under load. If it is not, not sending a token is probably ok. Better to save that 1 RTT! Return `true` in that case.
2. If the server is under load, it might make sense to require a token (maybe based on the `net.Addr`, if you want to go fancy). That means:
    * returning `false` if `token == nil`
    * else: validating the token. Have a look at the default token validation functions in quic-go for inspiration: https://github.com/lucas-clemente/quic-go/blob/332473668a992ce372c54c353b2bdc24344cf46e/server.go#L241-L259

Reply:
@marten-seemann Thanks for the clarifications! That's really helpful.

Any chance we can/should reuse the default token validation function?

Reply:
@marten-seemann I made a small update to this function based on your feedback. I think it'd be awesome if that `quic.defaultAcceptToken()` function was exported so that we could just call it from here and reuse that logic correctly.

Thanks for the guidance/help!